loadfile/loadfile.go

```diff
@@ -0,0 +1,47 @@
+// Package loadfile provides functions to load files from a directory.
+package loadfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// LoadContext reads the contents at each file into a map where the key
+// is the absolute filepath and file contents is the value.
+func LoadContext(neededFilepaths []string) (map[string][]byte, error) {
+	result := map[string][]byte{}
+	var err error
+	for _, filePath := range neededFilepaths {
+		absPath, err := filepath.Abs(filePath)
+		if err != nil {
+			return nil, fmt.Errorf("error loading context files, failed to obtain absolute path of file %s (%w)",
+				filepath.Base(filePath), err)
+		}
+		fileData, err := os.ReadFile(absPath) //nolint:gosec
+		if err != nil {
+			return nil, fmt.Errorf("error loading context files, failed to read file %s (%w)", absPath, err)
```

Comment:
Seems unnecessarily indirect. I'd have gone with something more like "error loading context file %s (%w)". Although a better diagnosis (you don't currently retain enough information to do this easily, though) would be "error loading workflow file %s" (or config file, etc.: i.e., your `requiredFiles` keys.)